code/modules/logging/log_entry.dm

```diff
@@ -99,19 +99,37 @@ GENERAL_PROTECT_DATUM(/datum/log_entry)
 
 #undef MANUAL_JSON_ENTRY
 
+#if DM_VERSION >= 515
+#define CHECK_AND_TRY_FILE_ERROR_RECOVERY(file) \
+	var/static/in_error_recovery = FALSE; \
+	if(!fexists(##file)) { \
+		if(in_error_recovery) { \
+			in_error_recovery = FALSE; \
+			CRASH("Failed to error recover log file: [file]"); \
+		}; \
+		in_error_recovery = TRUE; \
+		logger.Log(LOG_CATEGORY_INTERNAL_ERROR, "attempting to perform file error recovery: [file]"); \
+		logger.init_log_category(logger.log_categories[category]?.type); \
+		call(src, UNLINT(__PROC__))(arglist(args)); \
+		return; \
+	}; \
+	in_error_recovery = FALSE;
+#else
+#define CHECK_AND_TRY_FILE_ERROR_RECOVERY(file)
```

Comment:
couldn't you give it it's old behavior of just crashing

Reply:
we're increasing min version to 515, so no